File: sdk/java/Message.java
package io.zarkwaf.messenger;

import com.sun.jna.Structure;
import com.sun.jna.Pointer;
import com.sun.jna.NativeLong;

@Structure.FieldOrder({"topic", "payload", "payloadLength"})
public class Message extends Structure {
    public Pointer topic;
    public Pointer payload;
    public NativeLong payloadLength;

    public static class ByValue extends Message implements Structure.ByValue {}
}

File: sdk/dotnet/Message.cs
using System.Runtime.InteropServices;

namespace ZarkWaf.Messenger
{
    [StructLayout(LayoutKind.Sequential)]
    internal struct Message
    {
        [MarshalAs(UnmanagedType.LPStr)]
        public string Topic;
        [MarshalAs(UnmanagedType.LPArray, SizeParamIndex = 2)]
        public byte[] Payload;
        public UIntPtr PayloadLength;
    }
} 